### PR TITLE
python3Packages.poetry-core: 1.0.0 -> 1.0.2

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.0.2

Change log:
- https://github.com/python-poetry/poetry-core/releases/tag/1.0.2
- https://github.com/python-poetry/poetry-core/releases/tag/1.0.1

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/check-manifest/default.nix [hunk 1]
```diff
@@ -14,13 +24,21 @@ buildPythonPackage rec {
...
   '';
 
-  propagatedBuildInputs = [ build pep517 toml ];
+  propagatedBuildInputs = [
+    build
+    pep517
+    toml
+  ];
 
-  checkInputs = [ mock breezy git pytestCheckHook ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/check-manifest/default.nix]: The commit message doesn't make sense. It is already using `pytestCheckHook`.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114970` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.poetry-core</li>
<li>python38Packages.poetry-core</li>
<li>python39Packages.poetry-core</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
python37Packages.poetry-core:
<details>


warning: unused-argument
Unused argument: `isort`.
Near pkgs/development/python-modules/poetry-core/default.nix:4:3:

```
  |
4 | , isort
  |   ^
```
</details>
</details>

File: pkgs/development/python-modules/check-manifest/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/check-manifest/default.nix]: The commit message made sense before it was rebased against `staging-next`.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 114970` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python3Packages.poetry-core</li>
  </ul>
</details>

File: pkgs/development/python-modules/check-manifest/default.nix [hunk 2]
```diff
@@ -14,9 +24,20 @@ buildPythonPackage rec {
     substituteInPlace tests.py --replace "test_build_sdist" "no_test_build_sdist"
   '';
 
-  propagatedBuildInputs = [ build pep517 toml ];
```
Comment (FRidh) [inline pkgs/development/python-modules/check-manifest/default.nix:17]: I found it surprising to see that `build` is a runtime dependency. So `check-manifest` is a build-time dependency only, hence it makes more sense it executes `build`. `check-manifest` should typically be in `nativeBuildInputs` in other expressions.

**Comment by FRidh** [comment]
Thank you

**Pull request merged**